pkg/sql/sem/tree/type_check.go

```diff
@@ -2607,6 +2610,24 @@ func checkTupleHasLength(t *Tuple, expectedLen int) error {
 	return nil
 }
 
+// isNullOrAnnotatedNullTuple returns true if the given expression is a DNull or
+// a DNull that is wrapped by an AnnotateTypeExpr with a tuple type.
+func isNullOrAnnotatedNullTuple(
```

Comment:
do you know nulls annotated as other types don't cause a problem? if so, maybe explain here or in the commit message.